shared/src/main/scala/bloop/io/Environment.scala

```diff
@@ -0,0 +1,174 @@
+package bloop.io
+
+object Environment {
+
+  /**
+   * Returns "\n" if both OSTYPE and SHELL are nonEmpty, returns `System.lineSeparator` otherwise.
+   * @see [[shellLineSeparator]] which is similar.
+   */
+  def lineSeparator: String = if (validOstype.nonEmpty) shellLineSeparator else System.lineSeparator
+
+  /**
+   * Returns preferred SHELL environment lineSeparator:
+   *    '\r\n' on Windows, unless SHELL environment is recognized.
+   *    '\n'   everywhere else
+   *
+   * The default ending in Windows is '\r\n', but either of two conditions can
+   * change the value of [[shellLineSeparator]]:
+   *    1. if [[validShell]] is nonEmpty
+   *    2. if system property `"line.separator"` is modified
+   *
+   * Unlike [[lineSeparator]], only depends on the value of SHELL.
+   *
+   * @return preferred lineSeparator for SHELL environments, or `System.lineSeparator` otherwise.
+   */
+  def shellLineSeparator: String = if (validShell.nonEmpty) "\n" else sysLineSeparator
+
+  /**
+   * Preferred line ending for SHELL environments.
+   *
+   * An alias for `shellLineSeparator`.
+   *
+   * @example
+   * {{{
+   *   print(someText + EOL)
+   * }}}
+   */
+  lazy val EOL: String = shellLineSeparator // alias for shellLineSeparator
```

Comment:
```suggestion
  lazy val EOL: String = shellLineSeparator
```
it's already in the scaladoc above.